### ISSUE TITLE
Incorrect ARG warning in multistage dockerfile

### ISSUE THREAD

**Issue by ngan** [RedCoolBeans/dockerlint#66]
### Issue details

If you have an `ARG` that you'd like to be shared for all stages, the [convention is to do the following](https://github.com/moby/moby/issues/37345#issuecomment-400245466):

```
# This is global.
ARG FOO=/path/to/something

FROM alpine:latest as base1
# Call `ARG` with just the key to load the variable from global.
ARG FOO
COPY $FOO/bar .
WORKDIR $FOO
```

The linter will then say:
```
COPY contains undefined ARG or ENV
WORKDIR path $FOO must be absolute
```

### Steps to reproduce/test case

See description above

### Affected version(s)

`master`
